controllers/bucket_controller_test.go

```diff
@@ -80,3 +583,120 @@ func mockFile(root, path, content string) error {
 	}
 	return nil
 }
+
+type s3MockObject struct {
+	Key          string
+	LastModified time.Time
+	ContentType  string
+	Content      []byte
+}
+
+type s3MockServer struct {
```

Comment:
Very interesting and useful s3 mock server. :+1:

Reply:
Forgotten work from @relu in a PR that never got merged :-)